### PR TITLE
Yf/bulkupload

### PR DESCRIPTION
**Pull request by yunfangjuan**
bulk loader works but probably needs more error handling.

### CODE REVIEW

File: app/assets/javascripts/components/bulkUploadImport.jsx [hunk 1]
```diff
@@ -0,0 +1,399 @@
...
+        that.initializeSelectTag();
+      });
+    })
+    .catch((error) => {
+      that.setState({
+        invalid: true,
+        errorMessage: 'Project exists already or is invalid'
+      })
+    });
+  }
```
Comment (boris-dimitrov) [inline app/assets/javascripts/components/bulkUploadImport.jsx]: very nice & ajaxy thanks

File: app/assets/javascripts/components/bulkUploadImport.jsx [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+
+  isProjectInvalid() {
+    if (this.refs.new_project.value === '' && this.state.project === 'Select a project') {
+      this.setState({
+        invalid: true,
+        errorMessage: 'Please enter valid project name'
+      })
+      return true;
+    } else {
+      return false;
```
Comment (boris-dimitrov) [inline app/assets/javascripts/components/bulkUploadImport.jsx]: it just seems a bit unexpected for a predicate function to modify state, especially only in one of the cases

File: app/assets/javascripts/components/bulkUploadImport.jsx (hunk 1 above)
Comment (yunfangjuan) [reply app/assets/javascripts/components/bulkUploadImport.jsx]: actually it's susan's code.

Comment (boris-dimitrov) [reply app/assets/javascripts/components/bulkUploadImport.jsx]: great @sadelokiki

**Pull request merged**